src/diffusers/pipelines/flux/pipeline_flux.py

```diff
@@ -465,15 +483,18 @@ def prepare_latents(
         generator,
         latents=None,
     ):
+        if latents is not None:
+            if latents.ndim == 4:
+                # Packing the latents to be of shape (batch_size, num_patches, channels)
+                latents = self._pack_latents(latents, batch_size, num_channels_latents, height, width)
```

Comment:
Is this check necessary? `check_inputs` will raise an error if an `ndim == 4` latent tensor is passed in? Also `_prepare_latent_image_ids` divides the height and width by 2 before creating the image ids? if the height isn't scaled up before passing into this method, you will get an incorrect shape.

Reply:
Perhaps checking the number of dimensions is too harsh. However, there must be a way to clearly tell the user the correct shapes for the latents. This was my issue when trying to use the model with custom latents and coming from other "traditional" models. The only thing to guide me before going into the source code was a size error mismatch. That is, in other models, I'm used to doing the following:

```py
pipe = ...
# ...
latents_shape = (
    batch_size,
    pipe.transformer.config.in_channels,
    height // pipe.vae_scale_factor,
    width // pipe.vae_scale_factor,
)
latents = torch.randn(latents_shape, ...)
# ...
image = pipe(
    prompt=prompt,
    # ...
    latents=latents
).images[0]
```

For a similar code to work with this model, I had to do it like so:

```py
pipe = ...
# ...
latents_shape = (
    batch_size,
    pipe.transformer.config.in_channels // 4,
    2 * height // pipe.vae_scale_factor,
    2 * width // pipe.vae_scale_factor,
)
latents = torch.randn(latents_shape, ...)
latents = pipe._pack_latents(latents, *latents.shape)  # Note: 2x the original height and width!
# ...
image = pipe(
    prompt=prompt,
    # ...
    latents=latents
).images[0]
```
Hence, no issues were encountered when we would run `_prepare_latent_image_ids`. A distinction on the dimensions should be made: `height` and `width` generally refer to the image's dimensions, but in `latent_image_ids = self._prepare_latent_image_ids(batch_size, height, width, device, dtype)`, for example, the `height` and `width` refer to the *upscaled* latent dimensions, as you note.

I think either a method to generate the latents of the correct shape is warranted, or as I was doing in this PR of improving the docstrings and perhaps simply raising an error in order to avoid any mixup of dimensions.